keras_nlp/models/whisper/whisper_audio_feature_extractor.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+from tensorflow import keras
+
+try:
+    import librosa
+except ImportError:
+    librosa = None
+
+
+NUM_MELS = 80
+
+
+class WhisperAudioFeatureExtractor(keras.layers.Layer):
+    """
+    Whisper audio feature extractor layer.
+
+    This layer takes in a batch of audio tensors, and computes the log-mel
+    spectrogram features for each audio tensor.
+
+    Args:
+        sample_rate: int, defaults to 16000. The sample rate of the audio.
+        n_fft: int, defaults to 400. The size of the Fourier Transform in STFT.
+        hop_length: int, defaults to 160. The distance between neighboring
+            sliding window frames while computing STFT.
+        chunk_length: int, defaults to 30. The length of each audio chunk in
```

Comment:
Is this the sequence length of the output sequences? If so we can use something like `output_sequence_length` (already in use in related layers)

Reply:
Not exactly. The audio is padded/trimmed to length `chunk_length * sample_rate` before computing the log-mel spectrogram. I've added a note on the output shape:

```
    The input audio tensor can either be of shape `(length_of_audio,)` or
    `(batch_size, length_of_audio)`. The output is a tensor of shape
    `(batch_size, num_frames, NUM_MELS)`, where `num_frames` is
    `length_of_audio / stride` and `NUM_MELS` is 80.
```